evap/staff/tests/test_importers.py

```diff
@@ -175,30 +175,38 @@ def test_enrollment_importer_high_enrollment_warning(self):
         self.assertIn("Warning: User bastius.quid@external.example.com has 3 enrollments, which is a lot.", warnings_many)
 
     def test_random_file_error(self):
+        semester = mommy.make(Semester)
         original_user_count = UserProfile.objects.count()
 
         with open(self.filename_random, "rb") as excel_file:
             excel_content = excel_file.read()
 
-        __, __, __, errors_test = UserImporter.process(excel_content, test_run=True)
-        __, __, __, errors_no_test = UserImporter.process(excel_content, test_run=False)
+        __, __, errors_test = EnrollmentImporter.process(excel_content, semester, None, None, test_run=True)
+        __, __, errors_no_test = EnrollmentImporter.process(excel_content, semester, None, None, test_run=False)
 
         self.assertEqual(errors_test, errors_no_test)
         self.assertIn("Couldn't read the file. Error: Unsupported format, or corrupt file:"
                 " Expected BOF record; found b'42\\n'", errors_test)
         self.assertEqual(UserProfile.objects.count(), original_user_count)
 
     def test_invalid_file_error(self):
+        semester = mommy.make(Semester)
+        mommy.make(CourseType, name_de="Seminar")
+        mommy.make(CourseType, name_de="Vorlesung")
         original_user_count = UserProfile.objects.count()
 
         with open(self.filename_invalid, "rb") as excel_file:
             excel_content = excel_file.read()
 
-        __, __, __, errors_test = UserImporter.process(excel_content, test_run=True)
-        __, __, __, errors_no_test = UserImporter.process(excel_content, test_run=False)
+        __, __, errors_test = EnrollmentImporter.process(excel_content, semester, None, None, test_run=True)
+        __, __, errors_no_test = EnrollmentImporter.process(excel_content, semester, None, None, test_run=False)
 
         self.assertEqual(errors_test, errors_no_test)
-        self.assertIn('Sheet "Sheet1", row 2: Email address is missing.', errors_test)
```

Comment:
We may want to merge this with staff/test/test_views.py:573 (TestSemesterImportView:test_error_handling) since they are now basically testing the same thing? Would require some kind of import mocking though, don't know if it would make the code any easier.

Reply:
i didn't even know the other tests exist. yes, we should somehow deduplicate this and i'm open for suggestions. maybe it would be sufficient for the view tests to assert that some error is shown without checking the actual text while the importer tests make sure the correct errors are returned from the importer?